doc/ublk.1

```diff
@@ -0,0 +1,273 @@
+.PP
+\fB\-t, \-\-type\fR
+.RS 4
+Specifies the type of device to create\&. The three types of supported devices are null, loop and bnd\&.
+.RE
+.PP
+\fB\-n, \-\-number\fR
+.RS 4
+Create a device with this id\&. The device node will be /dev/ublkb_n_\&.
+.RE
+.PP
+\fB\-q, \-\-queue\fR
+.RS 4
+Number of queues to create\&. Each queue is services by a dedicated child process\&. Default is 1\&.
+.RE
+.PP
+\fB\-d, \-\-depth\fR
+.RS 4
+Maximum queue\-depthfor each queue\&. Default is 4096\&.
+.RE
+.PP
+\fB\-z, \-\-zero_copy\fR
+.RS 4
+zero copy requires 4k block size, and can remap ublk driver\*(Aqs io request into ublksrv\*(Aqs vm space
+.RE
+.PP
+\fB\-u, \-\-uring_comp\fR
+.RS 4
+Force to complete io cmd via io_uring_cmd_complete_in_task so that performance comparison is done easily with using task_work_add\&.
+.RE
+.PP
+\fB\-g, \-\-need_get_data\fR
+.RS 4
+User should issue io cmd again for write requests to set io buffer address and copy data from bio vectors to the userspace io buffer\&.
+.RE
+.PP
+\fB\-r, \-\-user_recovey\fR
+.RS 4
+Block devices are recoverable if ublk server exits and restarts\&. Outstanding I/O when ublk server exits is met with errors\&. I/O issued while there is no ublk server queues\&.
+.RE
+.PP
+\fB\-i, \-\-user_recovey_reissue\fR
+.RS 4
+Block devices are recoverable if ublk server exits and restarts Outstanding I/O when ublk server exits is reissued I/O issued while there is no ublk server queues
+.RE
+.PP
+\fB\-e, \-\-user_recovey_fail_io\fR
+.RS 4
+Block devices are recoverable if ublk server exits and restarts Outstanding I/O when ublk server exits is met with errors I/O issued while there is no ublk server is met with errors
+.RE
+.PP
+\fB\-\-debug_mask\fR
+.RS 4
+Bitmask specifying which debug features to enable\&.
+.RE
+.PP
+\fB\-\-unprivileged\fR
+.RS 4
+Unprivileged user can create /dev/ublkcN and /dev/ublkbN\&.
+.sp
+/dev/ublk\-control needs to be available for unprivileged user, and it can be done via udev rule to make all control commands available to unprivileged user\&. Except for the command of UBLK_CMD_ADD_DEV, all other commands are only allowed for the owner of the specified device\&.
+.sp
+When userspace sends UBLK_CMD_ADD_DEV, the device pair\*(Aqs owner_uid and owner_gid are stored to ublksrv_ctrl_dev_info by kernel, so far only the current user\*(Aqs uid/gid is stored, that said owner of the created device is always the current user\&.
+.sp
+We still need udev rule to apply OWNER/GROUP with the stored owner_uid and owner_gid\&.
+.sp
+Then ublk server can be run as unprivileged user, and /dev/ublkbN can be accessed and managed by its owner represented by owner_uid/owner_gid\&.
+.RE
+.PP
+\fB\-\-user_copy\fR
+.RS 4
+Copy between request and user buffer by pread()/pwrite()
+.RE
+.PP
+\fB\-\-eventfd\fR
```

Comment:
--eventfd is one internal switch, and I think it needn't to be documented in manpage.